File: backend/middleware/authorize.js
const authorize = (roles = []) => {
  if (typeof roles === "string") {
    roles = [roles];
  }

  return (req, res, next) => {
    if (roles.length && !roles.includes(req.userRole)) {
      return res.status(403).json({ message: "Acceso denegado." });
    }
    next();
  };
};

module.exports = authorize;

File: audit-service/middleware/authorize.py
from functools import wraps
from flask import request, jsonify

def authorize(roles=None):
    if isinstance(roles, str):
        roles = [roles]
    
    def decorator(f):
        @wraps(f)
        def decorated_function(*args, **kwargs):
            user_role = getattr(request, 'user_role', None)
            if user_role not in roles:
                return jsonify({"message": "Acceso denegado."}), 403
            return f(*args, **kwargs)
        return decorated_function
    return decorator
